docs/development/frameworks/apache-tomcat-on-centos-8/index.md

```diff
@@ -0,0 +1,140 @@
+[Unit]
+Description=Tomcat Server
+After=syslog.target network.target
+
+[Service]
+Type=forking
+User=tomcat
+Group=tomcat
+
+Environment=JAVA_HOME=/usr/lib/jvm/jre        Environment='JAVA_OPTS=-Djava.awt.headless=true'
+Environment=CATALINA_HOME=/usr/local/tomcat
+Environment=CATALINA_BASE=/usr/local/tomcat
+Environment=CATALINA_PID=/usr/local/tomcat/temp/tomcat.pid
+Environment='CATALINA_OPTS=-Xms512M -Xmx1024M'
+ExecStart=/usr/local/tomcat/bin/catalina.sh start
+ExecStop=/usr/local/tomcat/bin/catalina.sh stop
+
+[Install]
+WantedBy=multi-user.target
+{{< /file >}}
+
+1. Reload the `systemd` daemon to let it know about the `tomcat.service` that you created:
+
+        sudo systemctl daemon-reload
+
+1. Start and enable Tomcat server:
+
+        sudo systemctl enable tomcat
+        sudo systemctl start tomcat
+
+1. Configure your firewall to access Tomcat server on port 8080:
+
+        sudo firewall-cmd --permanent --zone=public --add-port=8080/tcp
+        sudo firewall-cmd --reload
+
+## Test and Use Tomcat
+
+You can test your Tomcat installation by pointing your browser at your site's port `:8080`, `http://example.com:8080/`, replacing `example.com` with your domain name. Note that Tomcat listens on network port 8080 and does not accept forced HTTPS connections by default. By default, Tomcat configuration files are located in the `/usr/local/tomcat/conf` directory.
+
+### Configure tomcat9-admin (optional)
+
+1.  To use the `tomcat9-admin` web application, add the following lines to the end of your `/usr/local/tomcat/conf/tomcat-users.xml` file before the `</tomcat-users>` line, substituting your own username and secure password. If using Tomcat Admin, include both the "manager-gui" role for the manager and the "admin-gui" role for the host-manager application.
+
+    {{< file "/usr/local/tomcat/conf/tomcat-users.xml" xml >}}
+<role rolename="manager-gui"/>
+<role rolename="admin-gui"/>
+<user username="username" password="password" roles="manager-gui,admin-gui"/>
+
+{{< /file >}}
+
+    {{< note >}}
+If you are not using the web application and plan to manage your application(s) from the command line only, you should not enter these lines, because doing so may expose your server to unauthorized login attempts.
+{{</ note >}}
+
+1.  For Tomcat versions 8+ the managers have been pre-configured to only allow access from the same IP of the server where it's installed. If you're trying to access it from a browser remotely, you'll need to comment out this configuration in the file `/usr/local/tomcat/webapps/manager/META-INF/context.xml`.
+
+    {{< file "/usr/local/tomcat/webapps/manager/META-INF/context.xml" xml >}}
+...
+<!--
+  <Valve className="org.apache.catalina.valves.RemoteAddrValve"
+         allow="127\.\d+\.\d+\.\d+|::1|0:0:0:0:0:0:0:1" />
+-->
+...
+{{</ file >}}
+
+1.  Restart the Tomcat server, which will allow these changes to take effect:
+
+        sudo systemctl restart tomcat
```

Comment:
I added this section because I wasn't able to get a login prompt for these managers. With some research I found that with later versions of tomcat they are restricting these pages for local installation only.